aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -222,17 +222,22 @@ Tensor addmm_sparse_csr_dense(
     const Tensor& dense,
     const Scalar& beta,
     const Scalar& alpha) {
-  Tensor r = at::empty({0}, self.options());
+  Tensor r = at::empty({0, 0}, self.options());
   at::addmm_out(r, self, sparse, dense, beta, alpha);
   return r;
 }
 
-SparseCsrTensor& _sparse_csr_mm_out(
-    const SparseCsrTensor& sparse,
-    const Tensor& dense,
-    SparseCsrTensor& result) {
-  Tensor t = at::zeros({}, dense.options());
-  return at::addmm_out(result, t, sparse, dense, 0.0, 1.0); // redispatch!
+Tensor& _sparse_csr_mm_out(
+    const Tensor& mat1,
+    const Tensor& mat2,
+    Tensor& result) {
+  Tensor zero;
+  if (result.is_sparse_csr()) {
+    zero = at::empty({mat1.size(0), mat2.size(1)}, mat2.options());
+  } else {
+    zero = at::zeros({mat1.size(0), mat2.size(1)}, mat2.options());
```

Comment:
I'm trying to think of a reason as to why we're not using empty for both.

Reply:
It should be zeros for both, not empty, but for sparse csr empty here is equivalent to a matrix full of zeros. `at::zeros` is not implemented for sparse csr yet.

Reply:
Wouldn't at::zeros for Sparse just be an entirely empty Tensor, i.e. values, crow and col indices all have 0 numel and only the size is set?